### ISSUE TITLE
Clustering

### ISSUE THREAD

**Issue by Erfan1369** [aertslab/pySCENIC#430]
Hello SCENIC Developers,

I have a single-cell datasets which has been analyzed up to the end of clustering step using Seurat(I have 8 clusters).
I was curious that is it reasonable to run Pyscenic on only one cluster or on a number of selected clusters (e.x 3 or 4 clusters which are biologically related)? and also, Is it necessary to perform clustering based the Pyscenic protocol?

Thanks!!

**Comment by Erfan1369** [comment]
@cflerin

**Comment by SeppeDeWinter** [comment]
Hi @Erfan1369 

Sorry for the late reply.

It is for sure possible to run pyscenic on only a subset of cells (e.g. one or a few clusters). However, take note that the pyscenic algorithm depends on heterogeneity/variability to be able to find correlations/co-expressiong. 

For example if the expression of a TF does not change across your single cluster you won't be able to find a regulon for that TF.

However this approach of running on a select set of clusters is a good idea in cases where your TF(s) might have different target genes in different cell types, by running on a select subset you might be able to disentangle this effect.

In a nutshell it depends on your dataset and biological question.

Hope this helps,

Best,

Seppe
